core/fetcher/fetcher.go

```diff
@@ -162,19 +169,19 @@ func (f *Fetcher) fetchAggregatorData(ctx context.Context, slot int64, defSet co
 			continue
 		}
 
-		// Query AggSigDB for DutyAttester to get attestation data root.
-		attData, err := f.aggSigDBFunc(ctx, core.NewAttesterDuty(slot), pubkey)
+		// Query DutyDB for Attestation data to get attestation data root.
+		attData, err := f.awaitAttDataFunc(ctx, slot, int64(res.CommitteeIndex))
 		if err != nil {
 			return core.UnsignedDataSet{}, err
 		}
 
-		att, ok := attData.(core.Attestation)
-		if !ok {
-			return core.UnsignedDataSet{}, errors.New("invalid attestation")
+		dataRoot, err := attData.HashTreeRoot()
+		if err != nil {
+			return core.UnsignedDataSet{}, err
 		}
 
 		// Query BN for aggregate attestation.
-		aggAtt, err := f.eth2Cl.AggregateAttestation(ctx, eth2p0.Slot(slot), att.Data.BeaconBlockRoot)
```

Comment:
@xenowits this was the bug, attestation data root is needed and NOT a beacon block root